src/Sulu/Bundle/CoreBundle/DependencyInjection/Compiler/RequestAnalyzerCompilerPass.php

```diff
@@ -22,63 +19,21 @@
  */
 class RequestAnalyzerCompilerPass implements CompilerPassInterface
 {
+    const TAG_NAME = 'sulu.request_attributes';
+
     /**
      * {@inheritdoc}
      */
     public function process(ContainerBuilder $container)
     {
-        // set website request analyzer
-        $container->setDefinition(
-            'sulu_core.webspace.request_analyzer.website',
-            new Definition(
-                'Sulu\Component\Webspace\Analyzer\WebsiteRequestAnalyzer',
-                [
-                    new Reference('sulu_core.webspace.webspace_manager'),
-                    $container->getParameter('kernel.environment'),
-                ]
-            )
-        );
-        // set admin request analyzer
-        $container->setDefinition(
-            'sulu_core.webspace.request_analyzer.admin',
-            new Definition(
-                'Sulu\Component\Webspace\Analyzer\AdminRequestAnalyzer',
-                [
-                    new Reference('sulu_core.webspace.webspace_manager'),
-                    $container->getParameter('kernel.environment'),
-                ]
-            )
-        );
-
-        if ($container->getParameter('sulu.context') === SuluKernel::CONTEXT_WEBSITE) {
-            $container->setAlias('sulu_core.webspace.request_analyzer', 'sulu_core.webspace.request_analyzer.website');
-        } else {
-            $container->setAlias('sulu_core.webspace.request_analyzer', 'sulu_core.webspace.request_analyzer.admin');
+        $references = [];
+        foreach ($container->findTaggedServiceIds(self::TAG_NAME) as $id => $tags) {
+            foreach ($tags as $attributes) {
+                $priority = array_key_exists('priority', $attributes) ? $attributes['priority'] : 0;
+                $references[$priority] = new Reference($id);
```

Comment:
The priority should be used as in the event dispatcher, i.e. The priority should be reversed and each priority must be assignable more than once.